flow360/component/project.py

```diff
@@ -0,0 +1,560 @@
+        Runs the surface mesher for the project.
+
+        Parameters
+        ----------
+        params : SimulationParams
+            Simulation parameters for running the mesher.
+        name : str, optional
+            Name of the draft (default is "SurfaceMesh").
+        run_async : bool, optional
+            Whether to run the mesher asynchronously (default is True).
+        fork : bool, optional
+            Whether to fork the case (default is False).
+
+        Raises
+        ------
+        Flow360ValueError
+            If the root item type is not Geometry.
+        """
+        self._check_initialized()
+        if self.metadata.root_item_type is not RootType.GEOMETRY:
+            raise Flow360ValueError(
+                "Surface mesher can only be run by projects with a geometry root asset"
+            )
+        self._surface_mesh = self._run(
+            params=params, target=SurfaceMesh, draft_name=name, run_async=run_async, fork=fork
+        )
+
+    @pd.validate_call
+    def generate_volume_mesh(
+        self,
+        params: SimulationParams,
+        name: str = "VolumeMesh",
+        run_async: bool = True,
+        fork: bool = True,
+    ):
+        """
+        Runs the volume mesher for the project.
+
+        Parameters
+        ----------
+        params : SimulationParams
+            Simulation parameters for running the mesher.
+        name : str, optional
+            Name of the draft (default is "VolumeMesh").
+        run_async : bool, optional
+            Whether to run the mesher asynchronously (default is True).
+        fork : bool, optional
+            Whether to fork the case (default is True).
+
+        Raises
+        ------
+        Flow360ValueError
+            If the root item type is not Geometry.
+        """
+        self._check_initialized()
+        if self.metadata.root_item_type is not RootType.GEOMETRY:
+            raise Flow360ValueError(
+                "Volume mesher can only be run by projects with a geometry root asset"
+            )
+        self._volume_mesh = self._run(
+            params=params,
+            target=VolumeMeshV2,
+            draft_name=name,
+            run_async=run_async,
+            fork=fork,
+        )
+
+    @pd.validate_call
+    def run_case(
+        self,
+        params: SimulationParams,
+        draft_name: str = "Case",
+        run_async: bool = True,
+        fork: bool = True,
+    ):
+        """
+        Runs a case for the project.
+
+        Parameters
+        ----------
+        params : SimulationParams
+            Simulation parameters for running the case.
+        draft_name : str, optional
+            Name of the draft (default is "Case").
+        run_async : bool, optional
+            Whether to run the case asynchronously (default is True).
+        fork : bool, optional
+            Whether to fork the case (default is True).
+        """
+        self._check_initialized()
+        self._case = self._run(
```

Comment:
What if a user runs a batch of cases by tweaking parameter? Then _case will get overwritten every time a new case is submitted? Can we instead create a draft_name (which IIRC is the case/VM/SM name) to asset(or just asset ID) map instead? @maciej-flexcompute what do you think?

Reply:
You're right. I was initially under the impression that these properties will hold the most recently opened assets (geometry, surface mesh, volume mesh, case) but in case of batch runs we should probably keep some history of all loaded asset objects... Maybe we could have a history of sorts where the user can see previous assets?

Something like

```
for i in range(0, 50):
    project.run_case(...)

# Show most recent one
case = project.case

# Show history of all project runs
for id in project.case_history():
    # Load the project with a specified ID
    case = project.case(id)
```

Reply:
I see. We need to fix this though since I expect the major use for python client is batch submission.

Another simple way is to just return the `Case` obj and ask user to manage their cases.